### ISSUE TITLE
App crashes when loading only one DRM

### ISSUE THREAD

**Issue by Chaphasilor** [HorizonLauncher/Client#75]
So when I start Horizon for the first time (or after DB reset), in the upcoming dialougue I select "Add DRM". It then correctly finds Steam but not Origin nor uPlay (which are both installed). When I then want to continue only with Steam the App crashes. That means I cannot add steam games to my libary 'cause you cannot add the .url of steam game shortcuts manually.

**Comment by EionRobb** [comment]
I think this is the same crash I'm getting.  I'm using the 44f9da7 snapshot on Windows and it gives the error:

```
Could not parse stylesheet of object 0x8ea5ee8
Parsing Steam vdf, located at: "C:/program files (x86)/steam/appcache/appinfo.vdf"
Assertion failed!

Program: HorizonLauncher.exe
File: C:/Boost/boost_1_58_0/boost/property_tree/detail/ptree_implementation.hpp, Line 905

Expression: !p.empty() && "Empty path not allowed for put_child."

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

**Comment by thomassross** [comment]
So it crashes when you click next on this page?

![page](http://i.imgur.com/CShEU18.png)

**Comment by Chaphasilor** [comment]
Hm, using same build its the following error for me:

```
Parsing Steam vdf, located at: "C:/program files (x86)/steam/appcache/appinfo.vd
f"
terminate called after throwing an instance of 'boost::exception_detail::clone_i
mpl<boost::exception_detail::error_info_injector<boost::property_tree::ptree_bad
_path> >'
  what():  No such node (AppState.installdir)

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

**Comment by Chaphasilor** [comment]
It's after this screen:

![untitled](https://cloud.githubusercontent.com/assets/18015147/13966285/e7322c0c-f071-11e5-856f-b31812115ab4.png)

**Comment by EionRobb** [comment]
Same for me

**Comment by thomassross** [comment]
Thanks. Looking into it now.

**Comment by thomassross** [comment]
Can you download the source code from [here](https://github.com/HorizonLauncher/Client/archive/fix-75.zip) and try [building from source](https://github.com/HorizonLauncher/Client/wiki/Building-Horizon-Launcher-on-Windows)? I've made some changes which _should_ fix the issue.

**Comment by Chaphasilor** [comment]
Ok just give me some minutes

**Comment by thomassross** [comment]
Alright, thanks.

**Comment by Chaphasilor** [comment]
Uhh, not at 7% downloading Qt...

**Comment by Chaphasilor** [comment]
So CMake couldn't run properly; produced the following error:

```
-- The C compiler identification is MSVC 19.0.23506.0
-- The CXX compiler identification is MSVC 19.0.23506.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:168 (find_package):
  By not providing "FindQt5WebKit.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Qt5WebKit", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5WebKit" with
  any of the following names:

    Qt5WebKitConfig.cmake
    qt5webkit-config.cmake

  Add the installation prefix of "Qt5WebKit" to CMAKE_PREFIX_PATH or set
  "Qt5WebKit_DIR" to a directory containing one of the above files.  If
  "Qt5WebKit" provides a separate development package or SDK, be sure it has
  been installed.

-- Configuring incomplete, errors occurred!
See also "C:/Users/Chaphasilor/Desktop/Horizon Launcher/build-Client-fix-75-Desktop_Qt_5_6_0_MSVC2015_64bit-Default/CMakeFiles/CMakeOutput.log".
```

**Comment by wjoe** [comment]
I have the same issue on Linux. I was able to build the fix-75 client posted above without errors, but this didn't fix the issue.

Steps taken:
Run set up wizard
Click Add DRM
Steam is detected correctly in /home/joe/.local/share/Steam
Click Next
Horizon hangs for a moment, then crashes

Terminal output:

```
[joe@sol build]$ ./HorizonLauncher 
QSqlError("", "", "")
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
Could not parse stylesheet of object 0x1c59e40
Could not parse stylesheet of object 0x1460320
Could not parse stylesheet of object 0x1460320
Could not parse stylesheet of object 0x1460a00
Could not parse stylesheet of object 0x1460a00
Could not parse stylesheet of object 0x16a95d0
Could not parse stylesheet of object 0x16a95d0
Could not parse stylesheet of object 0x16aaeb0
Could not parse stylesheet of object 0x16aaeb0
Could not parse stylesheet of object 0x147f370
Could not parse stylesheet of object 0x147f370
Could not parse stylesheet of object 0x1482af0
Could not parse stylesheet of object 0x1482af0
Could not parse stylesheet of object 0x1484a40
Could not parse stylesheet of object 0x1484a40
Could not parse stylesheet of object 0x16e1770
Could not parse stylesheet of object 0x16e1770
Could not parse stylesheet of object 0x16e42f0
Could not parse stylesheet of object 0x16e42f0
Could not parse stylesheet of object 0x160fbc0
Could not parse stylesheet of object 0x160fbc0
Could not parse stylesheet of object 0x16b6e00
Could not parse stylesheet of object 0x16b6e00
Could not parse stylesheet of object 0x167f000
Could not parse stylesheet of object 0x167f000
Could not parse stylesheet of object 0x1681ad0
Could not parse stylesheet of object 0x1681ad0
Could not parse stylesheet of object 0x17784e0
Could not parse stylesheet of object 0x17784e0
Could not parse stylesheet of object 0x177aee0
Could not parse stylesheet of object 0x177aee0
Could not parse stylesheet of object 0x177da70
Could not parse stylesheet of object 0x177da70
Could not parse stylesheet of object 0x1c59e40
Could not parse stylesheet of object 0x1c59e40
Could not parse stylesheet of object 0x1460320
Could not parse stylesheet of object 0x1460a00
Could not parse stylesheet of object 0x16a95d0
Could not parse stylesheet of object 0x16aaeb0
Could not parse stylesheet of object 0x147f370
Could not parse stylesheet of object 0x1482af0
Could not parse stylesheet of object 0x1484a40
Could not parse stylesheet of object 0x16e1770
Could not parse stylesheet of object 0x16e42f0
Could not parse stylesheet of object 0x160fbc0
Could not parse stylesheet of object 0x16b6e00
Could not parse stylesheet of object 0x167f000
Could not parse stylesheet of object 0x1681ad0
Could not parse stylesheet of object 0x17784e0
Could not parse stylesheet of object 0x177aee0
Could not parse stylesheet of object 0x177da70
Could not parse stylesheet of object 0x1c59e40
Parsing Steam vdf, located at: "/home/joe/.local/share/Steam/appcache/appinfo.vdf"
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::property_tree::ptree_bad_path> >'
  what():  No such node (AppState.installdir)
Aborted (core dumped)
```

The errors regarding stylesheets and everything above that occur when starting Horizon, just including in case it's relevant. The lines below that occur when it crashes.

**Comment by thomassross** [comment]
@Chaphasilor are you running CMake with the CMake program or through QtCreator? You need to run it through Qt Creator.

@wjoe Do you mind sending me your `/home/joe/.local/share/Steam/appcache/appinfo.vdf` file? You can either attach it here or email it to me: t@thomasross.io

**Comment by Chaphasilor** [comment]
@thomassross I'm running it through Qt

**Comment by thomassross** [comment]
@Chaphasilor can you go into the `Qt Maintenance Tool` and make sure QtWebKit is installed?

**Comment by Chaphasilor** [comment]
@thomassross Where can I find that tool?

**Comment by thomassross** [comment]
@Chaphasilor It should've been installed when you installed Qt, just search for it.

**Comment by Chaphasilor** [comment]
Nope, not in my start menu whith the other Qt programms

Ok, found it now

**Comment by Chaphasilor** [comment]
Hmm, so I can't seem to find the missing "Qt5WebKit":

![untitled](https://cloud.githubusercontent.com/assets/18015147/13991561/fa0e2838-f118-11e5-934a-7ba8a5564504.png)

**Comment by thomassross** [comment]
Hi,

Please uninstall Qt 5.6 and install Qt 5.5 and try compiling.

**Comment by thomassross** [comment]
Hi,

This bug will be fixed when #28 is fixed. I am closing it.
